[develop] feat: initial AEROMMA use case implementation and MELODIES MONET integration (#1315)
ufs-community/ufs-srweather-app: doc/UsersGuide/BuildingRunningTesting/AQM.rst

.. code-block:: console

   $ ./run_we2e_tests.py -t my_tests.txt -m hera -a gsd-fv3 -q

AQM Use Cases
=============

> An AQM "use case" is a scientifically interesting air quality event with preconfigured SRW workflow templates. ICs, LBCs, and fixed files are also staged in the `SRW App Data Bucket <https://registry.opendata.aws/noaa-ufs-shortrangeweather/>`__, and can be downloaded using the `AQM-Eval Data Sync utility <https://github.com/NOAA-EPIC/AQM-Eval?tab=readme-ov-file#installation>`__. If data is downloaded using this utility, setting ``cpl_aqm_parm.USE_AQM_S3_DATA_STAGE = true`` and  ``AQM_STAGE_DST_DIR = <stage_directory>`` will populate the root paths for data dependencies.

Reviewer: The `Doc Tests` are failing because the test is unable to connect to `https://github.com/NOAA-EPIC/AQM-Eval?tab=readme-ov-file#installation` (anchor installation is not found).  It would be interesting to see if the test will work properly with the following URL - `https://github.com/NOAA-EPIC/AQM-Eval/blob/develop/README.md#installation`

```suggestion
An AQM "use case" is a scientifically interesting air quality event with preconfigured SRW workflow templates. ICs, LBCs, and fixed files are also staged in the `SRW App Data Bucket <https://registry.opendata.aws/noaa-ufs-shortrangeweather/>`__, and can be downloaded using the `AQM-Eval Data Sync utility <https://github.com/NOAA-EPIC/AQM-Eval/blob/develop/README.md#installation>`__. If data is downloaded using this utility, setting ``cpl_aqm_parm.USE_AQM_S3_DATA_STAGE = true`` and  ``AQM_STAGE_DST_DIR = <stage_directory>`` will populate the root paths for data dependencies.
```

Reviewer: This modification won't allow the Doc Tests to pass, so I will close the comment out.

.. list-table:: Supported Use Cases
   :widths: 20 10 20
   :header-rows: 1

   * - Name
     - Key
     - Configuration Template
   * - `Atmospheric Emissions and Reactions Observed from Megacities to Marine Areas <https://csl.noaa.gov/projects/aeromma/>`__
     - AEROMMA
     - ``./ush/aqm-use-cases/config.aqm.AEROMMA.yaml``

.. _acquire-use-case-data:

Acquiring Use Case Data
---------------------------

AQM data requirements are relatively large. Using the AEROMMA campaign as an example, expect a minimum of ~28 terabytes of storage for the full use case, with ~3 terabytes allocated to fixed files. Hence, use case configurations assume a user will want to evaluate a 48-hour forecast before extending to the recommended use case forecast window. In this situation, download the time-varying data using the data sync utility's ``--snippet`` flag.

.. code-block:: console

   $ DST_DIR=<path to root directory for sync>

>    $ conda run -n aqm-eval --no-capture-output aqm-data-sync time-varying --dst-dir ${DST_DIR} --use-case AEROMMA --snippet

Reviewer: In order to pull data for the default 48-hour forecast, I needed to use the following command:

`conda run -n aqm-eval --no-capture-output aqm-data-sync time-varying --dst-dir ${DST_DIR} --first-cycle-date 2023060112 --last-cycle-date 2023060212`

Using:

`conda run -n aqm-eval --no-capture-output aqm-data-sync time-varying --dst-dir ${DST_DIR} --use-case AEROMMA --snippet`

only retrieved data for `2023060112`, not `2023060212`, causing the majority of tasks for `2023060212` to fail.

```suggestion
   $ conda run -n aqm-eval --no-capture-output aqm-data-sync time-varying --dst-dir ${DST_DIR} --first-cycle-date 2023060112 --last-cycle-date 2023060212
```

.. tip::

   The sync utility is resumable. Re-use the destination directory if a sync operation is canceled, interrupted, or additional temporal data is needed. Synchronization is a one-way sync from the S3 source.

Running a Use Case
----------------------

Once data is appropriately staged, the use case workflow configuration file may be used like any other SRW workflow configuration file. The preconfigured template assumes users have downloaded data using the data synchronization utility. If not, the user should move any use case-specific settings to the default AQM configuration file (e.g., cycle dates). Assuming the data stage is done correctly for the use case, users should generally focus on updating these configuration variables:

.. list-table:: Important Use Case Configuration Variables
   :widths: 20 50
   :header-rows: 1

   * - Configuration Variable
     - Description
   * - ``user.MACHINE``
     - Platform running the workflow.
   * - ``user.ACCOUNT``
     - Account holding the core hours.
   * - ``cpl_aqm_parm.AQM_STAGE_DST_DIR``
     - Required path to the destination (root) directory of the data synchronization operation.
   * - ``workflow.DATE_LAST_CYCL``

>      - Defaults to a single 24-hour forecast cycle.

Reviewer: The default in `config.aqm.AEROMMA.yaml` is two 24-hour forecast cycles (2023060112 and 2023060212).

```suggestion
     - Defaults to two 24-hour forecast cycles.
```

Check Availability of ICs and LBCs
++++++++++++++++++++++++++++++++++++

For the evaulation to work properly, at least a 48-hour forecast is required. It may be necessary to download additional IC and LBC datasets. See :ref:`acquire-use-case-data`.

Enable the MM Workflow Task Group
+++++++++++++++++++++++++++++++++++

Under ``workflow.tasksgroups`` in the experiment configuration, add or uncomment ``- parm/wflow/aqm_post_melodies_monet.yaml``.

Configure Paths to Observational Datasets
+++++++++++++++++++++++++++++++++++++++++++

For AirNow, set ``task_mm_prep.MM_OBS_AIRNOW_FN_TEMPLATE`` to the appropriate path. Wildcards may be used it the experiment's forecast window extends beyond a month.

Configure "Scorecard" Base Model (Optional)
+++++++++++++++++++++++++++++++++++++++++++++

MM provides a set of `scorecard evaluations <https://melodies-monet.readthedocs.io/en/stable/users_guide/supported_plots.html#id8>`__ used to compare a base and evaluation model run. Setting ``task_mm_prep.MM_BASE_MODEL_EXPT_DIR`` to a different SRW experiment root directory will enable scorecard plotting.

Transfer MM Output (Optional)
+++++++++++++++++++++++++++++++

> Unless overridden by ``task_mm_prep.MM_OUTPUT_DIR``, the MM evaluation output is written toe ``${EXPT_DIR}/mm_output``. Users may wish to transfer the PNG and CSV output to their local machine for simplified review.

Reviewer: to rather than toe.

```suggestion
Unless overridden by ``task_mm_prep.MM_OUTPUT_DIR``, the MM evaluation output is written to ``${EXPT_DIR}/mm_output``. Users may wish to transfer the PNG and CSV output to their local machine for simplified review.
```
